src/lib.rs

```diff
@@ -172,7 +172,9 @@ impl<F: PrimeField, PC: PolynomialCommitment<F>, D: Digest> Marlin<F, PC, D> {
         zk_rng: Option<&mut dyn RngCore>,
     ) -> Result<Proof<F, PC>, Error<PC::Error>>
     {
-        assert!(zk_rng.is_some() && zk || zk_rng.is_none() && !zk);
+        if zk_rng.is_some() && !zk || zk_rng.is_none() && zk {
+            Err(Error::Other("ZK parameters incorrect".to_owned()))?
```

Comment:
"If ZK is enabled, a RNG must be passed (and viceversa); conversely, if ZK is disabled, a RNG must NOT be passed (and viceversa)"